xc/device.ml

```diff
@@ -2682,6 +2698,60 @@ module Dm = struct
 
   (* the following functions depend on the functions above that use the qemu backend Q *)
 
+  let start_varstored ~xs ?(nvram=[]) ?(restore=false) task domid =
+    let path = !Xc_resources.varstored in
+    let name = "varstored" in
+    let vm_uuid = Xenops_helpers.uuid_of_domid ~xs domid |> Uuidm.to_string in
+    let reset_on_boot = match List.assoc "EFI-variables-on-boot" nvram with
+      | "persist" -> false
+      | "reset" -> true
+      | bad ->
+        sprintf "unsupported EFI-variables-on-boot %s, use 'persist' or 'reset'" bad
+        |> fun s -> Internal_error s
+        |> raise
+      | exception Not_found -> false
+    in
+    let default_backend = "xapidb" in
+    let backend = match List.assoc "EFI-variables-backend" nvram with
+      | x when x = default_backend -> x
+      | bad ->
+        sprintf "unsupported EFI-variables-backend %s, use 'xapidb'" bad
+        |> fun s -> Internal_error s
+        |> raise
+      | exception Not_found -> default_backend
+    in
+    let efivars_init = match List.assoc "EFI-variables" nvram with
+      | efivars when not restore ->
+        let efivars_init_path = efivars_init_path domid in
+        Unixext.write_string_to_file efivars_init_path efivars;
+        Some efivars_init_path
+      | ""  | _ -> None
+      | exception Not_found -> None
```

Comment:
Maybe log this and the case above?

Reply:
I'll add some more logging, it would certainly be useful to know that we are starting a guest in uefi mode vs bios mode, and whether we found any nvram data or not.